hubbard/hamiltonian.py

```diff
@@ -55,20 +54,25 @@ def __init__(self, TBHam, n=0, U=0.0, q=(0., 0.), nkpt=[1, 1, 1], kT=1e-5):
         self.H = TBHam.copy()
         self.H.finalize()
         self.geometry = TBHam.geometry
+        # So far we only consider either unpolarized or spin-polarized Hamiltonians
+        self.spins = min(2, self.H.spin.spins)
```

Comment:
See https://github.com/zerothi/sisl/commit/8ee0de851b52c4c117501045c6ccfd1b46fef6db.

Reply:
Yeah, regarding `spin.spins` probably that is bad wording. It should more likely be `spin.components/elements`, or?

Reply:
> Yeah, regarding `spin.spins` probably that is bad wording. It should more likely be `spin.components/elements`, or?

`spin.spins` comes from sisl. Would you consider to change that?

Reply:
Yes and no. I will retain it for backwards compatibility, but I think perhaps `Spin.size` is more appropriate. I have amended sisl to use `Spin.size` instead. If you have another suggestion, please let me know!

The idea is that it allows one to determine the number of spin-elements required to describe the spin-kind. I.e. a polarized requires 2 elements. Whereas non-collinear requires 4 reals, or 3 complex arrays. Here `Spin.spins` doesn't really make sense. So the current discussion led me to believe its name suggested another use-case.

Reply:
I'm not sure which is the best name for our variable here. Maybe `self.spinors` can be a bit confusing? Since we are not using directly spinors here, and actually is the total number of spin components what we need. On the other hand, I'm using `self.spins = min(2, self.H.spin.spins)` because we only allow for spin polarized or unpolarized Hamiltonians, but if we had the implementation for SOC or NC then what would the dimension be?

Reply:
Maybe then follow sisl nomenclature and define `self.size = min(2, self.H.spin.size)`?

Reply:
Yes I think that's the best :)

Reply:
Yes, the `spin.spinor` is the number of diagonal elements in your spin box, that is either 1 (non polarized) or 2 (polarized, NC and SOC).

However, `spin.spins` doesn't really mean "number of spin components". It rather means, "number of numerical values needed to represent the spin-box". Therefore I think that `spin.spins` is not the best wording. So if you variable is used to signify spinors (diagonal components in the spin-box), then simply using `spin.spinor` would suffice.

Reply:
Yes I agree. So I was thinking on renaming the variable to `self.size` following `sisl` notation, but this makes sense if the dimension of the problem (e.g. the dimension of the spin-densities) then is determined by the  "number of numerical values needed to represent the spin-box" rather than the number of "spin components", which is what I am not entirely sure (in that case I guess it would make more sense to use `self.spinors` as you both point out)...

Reply:
Yeah, with @tfrederiksen suggestion you might as well use `Spin.spinor` since that is the same. I don't know exactly how +U looks for NC/SOC.